lib/hash.js

```diff
@@ -0,0 +1,74 @@
+const crypto = require('crypto');
+
+const SALT_LENGTH = 64;
+const HASH_ITERATIONS = 10000;
+const HASH_LENGTH = 256;
+const CIPHER = 'sha512';
+const SEPARATOR = ':';
+
+function genSalt() {
+  return crypto.randomBytes(SALT_LENGTH).toString('hex');
+}
+
+/**
+ * @callback saltAndHashCallback
+ * @param {Error}
+ * @param {String} responseMessage
+ */
+
+/**
+ * Generates a random salt and encrypts the given password
+ * @param  {String}   pwd Password to be encrypted
+ * @param  {saltAndHashCallback} cb Node-style callback
+ */
+exports.saltAndHash = function(pwd, cb) {
+  const salt = genSalt();
+  crypto.pbkdf2(pwd, salt, HASH_ITERATIONS, HASH_LENGTH, CIPHER, function(err, hashed) {
```

Comment:
PBKDF2 is surely safer than simply salted hash (or multiple salted hash), but it exposes to DOS attack.
If we want to use it, we should implement an increasing delay policy for each failed login attempt. Or, if possible, use a CAPTCHA.